src/Native/Runtime/arm64/GcProbe.S

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <unixasmmacros.inc>
+#include "AsmOffsets.inc"
+
+//
+// Set the Thread state and wait for a GC to complete.
+//
+// Register state on entry:
+//  x4: thread pointer
+//  
+// Register state on exit:
+//  x4: thread pointer
+//  All other registers trashed
+//
+    .global RhpWaitForGCNoAbort
+
+    .macro WaitForGCCompletion
+        ldr         w2, [x4, #OFFSETOF__Thread__m_ThreadStateFlags]
+        tst         w2, #TSF_SuppressGcStress__OR__TSF_DoNotTriggerGC
+        bne         0f
+
+        ldr         x9, [x4, #OFFSETOF__Thread__m_pHackPInvokeTunnel]
```

Comment:
I do not see anything that sets `m_pHackPInvokeTunnel` before this point. I do not think this can work.

It may be a good idea to test this well: Run `for (;;) { GC.Collect(); Thread.Sleep(1); }` on one thread, and some mix of GC allocations and PInvokes with `SuppressGCTransition` attribute on other thread.